TFGCasper.thy

```diff
@@ -0,0 +1,55 @@
+theory TFGCasper
+
+imports Main HOL.Real CBCCasper ExampleProtocols
+
+begin
+
+(* Section 4.4: Casper the Friendly GHOST *)
+
+(* Definition 4.23: Blocks *)
+typedecl block
+
+type_synonym genesis = block
+
+type_synonym next_block_generator = "block \<Rightarrow> block set"
+
+datatype ghost_params = GHOST_params "genesis * next_block_generator"
```

Comment:
In case this is not intended: datatype constructor can have multiple arguments:
```thy
datatype ghost_params = GHOST_params genesis next_block_generator
```

Reply:
Hmm, multiple arguments pattern is reasonable for me.
We also use tupled one arugment pattern in the following place.
https://github.com/LayerXcom/cbc-casper-proof/blob/786cb6efc4859cf6048e9559011429992b1aa950/CBCCasper.thy#L33-L34
I think we should rewrite these to multiple arguments pattern.
@nrryuya Do you have any idea?

Reply:
Either is fine?

Reply:
@yudetamago For params definition, record type looks better for me.
```thy
record params =
  V :: validator set
  W :: weight
  C :: threshould
  ...
```

Reply:
@myuon 
Thanks! I'll fix that in #44.